src/internal/components/chart-legend/index.tsx

```diff
@@ -236,133 +489,79 @@ export const ChartLegend = ({
       onUnregisterActive={(element: HTMLElement) => onUnregisterActive(element, navigationAPI)}
     >
       <div
-        ref={containerRef}
         role="toolbar"
-        aria-label={legendTitle || ariaLabel}
-        className={clsx(testClasses.root, styles.root, {
-          [styles["root-side"]]: position === "side",
-        })}
+        aria-label={ariaLabel}
         onMouseEnter={() => (isMouseInContainer.current = true)}
         onMouseLeave={() => (isMouseInContainer.current = false)}
+        className={clsx({
+          [styles["legend-dual-group"]]: isDual,
+        })}
       >
-        {legendTitle && (
-          <Box fontWeight="bold" className={testClasses.title}>
-            {legendTitle}
+        {title && (
+          <Box
+            fontWeight="bold"
+            className={testClasses.title}
+            textAlign={position === "bottom-opposite" ? "right" : "left"}
+          >
+            {title}
           </Box>
         )}
-
         <div
           // The list element is not focusable. However, the focus lands on it regardless, when testing in Firefox.
           // Setting the tab index to -1 does fix the problem.
           tabIndex={-1}
+          ref={containerRef}
+          style={{
+            overflow: "auto",
```

Comment:
The inline styles can only be used when the values are computed dynamically. The `overflow: auto` should be added via classes instead